app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,54 @@
+module Api
+  module V1
+    class ReservationsController < ApplicationController
+      load_and_authorize_resource
+      before_action :set_reservation, only: %i[show update destroy]
+
+      def index
+        @reservations = current_api_v1_user.reservations
+        render json: @reservations
+      end
+
+      def show
+        render json: @reservation
+      end
+
+      def create
+        @reservation = current_api_v1_user.reservations.build(reservation_params)
+
+        if @reservation.save
+          render json: @reservation, status: :created
+        else
+          render json: { errors: @reservation.errors.full_messages }, status: :unprocessable_entity
+        end
+      end
+
+      def update
+        if @reservation.update(reservation_params)
+          render json: @reservation
+        else
+          render json: { errors: @reservation.errors.full_messages }, status: :unprocessable_entity
+        end
+      end
+
+      def destroy
+        @reservation.destroy
+        head :no_content
+      end
+
+      private
+
+      def set_reservation
+        @reservation = current_api_v1_user.reservations.find(params[:id])
+      rescue ActiveRecord::RecordNotFound
+        render json: { error: 'Record not found' }, status: :not_found
+      end
```

Comment:
Update it here also so that we can get the same data `show` too.
```suggestion
     def set_reservation
        @reservation = current_api_v1_user.reservations.includes(:car).find(params[:id])
      rescue ActiveRecord::RecordNotFound
        render json: { error: 'Record not found' }, status: :not_found
      end
```

Reply:
@hamidazim321, That was unnecessary. However, if you don't want to follow the DRY principle and add unnecessary code to it, it's totally up to you."